sourcetool/pkg/sourcetool/implementation.go

```diff
@@ -2,16 +2,64 @@ package sourcetool
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/carabiner-dev/github"
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/sirupsen/logrus"
+	kgithub "sigs.k8s.io/release-sdk/github"
 
 	"github.com/slsa-framework/slsa-source-poc/sourcetool/pkg/attest"
 	"github.com/slsa-framework/slsa-source-poc/sourcetool/pkg/ghcontrol"
+	"github.com/slsa-framework/slsa-source-poc/sourcetool/pkg/policy"
 	"github.com/slsa-framework/slsa-source-poc/sourcetool/pkg/slsa"
 )
 
+const (
+	tokenVar = "GITHUB_TOKEN" //nolint:gosec // This are note creds, just the name
+
+	workflowPath   = ".github/workflows/compute_slsa_source.yaml"
+	workflowSource = "git+https://github.com/slsa-"
+)
+
+// TODO(puerco): Read this from latest version on the repository
+var workflowData = `# SPDX-FileCopyrightText: Copyright 2025 The SLSA Authors
+# SPDX-License-Identifier: Apache-2.0
+---
+name: SLSA Source
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
So this needs to match the protected branches in the policy we create.

If they're setting up for main, great.  If they're not using main or they're trying to protect something else it will be a problem.

We could just run for all branches if we wanted to make it easier on everyone.  (would be harder on GitHub).

I think things without policies should just default to L1.

If we don't want to fix this now we could handle it later and say we only support 'main' for now.

Reply:
Ah thanks, yes I actually forgot to implement this. I want to embed the workflow file (instead of holding it in a variable) so I just offloaded it from my mind to do it later. Fixed.